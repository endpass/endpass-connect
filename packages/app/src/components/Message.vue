<template>
  <p :class="{ message: true, error: error, ellipsis: ellipsis }">
    <slot/>
  </p>
</template>

<script>
export default {
  name: 'Message',

  props: {
    ellipsis: {
      type: Boolean,
      default: false
    },

    error: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style lang="postcss">
.message {
  font-size: 1em;

  &.error {
    color: #f00;
  }

  &.ellipsis {
    width: 100%;
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
  }
}
</style>
