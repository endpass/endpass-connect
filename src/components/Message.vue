<template>
  <p
    :class="{
      message: true,
      error: error,
      success: success,
      ellipsis: ellipsis,
    }"
  >
    <slot />
  </p>
</template>

<script>
export default {
  name: 'Message',

  props: {
    ellipsis: {
      type: Boolean,
      default: false,
    },

    error: {
      type: Boolean,
      default: false,
    },

    success: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style lang="postcss">
.message {
  font-size: 1em;
  line-height: 1.5;

  &.error {
    color: #f00;
  }

  &.success {
    color: #4dce4d;
  }

  &.ellipsis {
    width: 100%;
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
  }
}
</style>
