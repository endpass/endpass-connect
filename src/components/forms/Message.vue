<template>
  <div>
    <form-field>
      <message data-test="form-message">{{ message }}</message>
    </form-field>
    <form-controls>
      <v-button
        :disabled="!closable"
        data-test="cancel-button"
        @click="emitCancel"
        >Close</v-button
      >
    </form-controls>
  </div>
</template>

<script>
import VButton from '../VButton.vue';
import Message from '../Message.vue';
import FormField from '../FormField.vue';
import FormControls from '../FormControls.vue';

export default {
  name: 'MessageForm',

  props: {
    message: {
      type: String,
      required: true,
    },

    closable: {
      type: Boolean,
      default: false,
    },
  },

  methods: {
    emitCancel() {
      this.$emit('cancel');
    },
  },

  components: {
    VButton,
    Message,
    FormField,
    FormControls,
  },
};
</script>
