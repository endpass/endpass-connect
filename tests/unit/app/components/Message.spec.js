import { shallowMount } from '@vue/test-utils';
import Message from '@@/app/src/components/Message.vue';

describe('Message', () => {
  describe('render', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowMount(Message);
    });

    it('should correctly render Message component', () => {
      expect(wrapper.name()).toBe('Message');
      expect(wrapper.classes()).toEqual(['message']);
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should add error modifier if error props was passed in', () => {
      wrapper = shallowMount(Message, {
        propsData: {
          error: true,
        },
      });

      expect(wrapper.classes()).toContain('error');
    });

    it('should add ellipsis modifier if ellipsis props was passed in', () => {
      wrapper = shallowMount(Message, {
        propsData: {
          ellipsis: true,
        },
      });

      expect(wrapper.classes()).toContain('ellipsis');
    });
  });
});
