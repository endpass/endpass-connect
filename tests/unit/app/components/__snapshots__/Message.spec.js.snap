// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message render should correctly render Message component 1`] = `<p class="message"></p>`;
